src/Framework/AspNetCoreAnalyzers/test/Mvc/DetectAmbiguousActionRoutesTest.cs

```diff
@@ -318,6 +318,284 @@ static void Main(string[] args)
         await VerifyCS.VerifyAnalyzerAsync(source);
     }
 
+    [Fact]
+    public async Task ActionRouteToken_DifferentActionNames_NoDiagnostics()
```

Comment:
Should we add a test that covers one route using `[action]` and another using `{action}`?

Reply:
What kind of sick person would do that? I love it.

Reply:
I looked at this and the test isn't necessary. `HttpGet("[action]")` and `HttpGet("{action}")` aren't flagged as ambiguous because the routes are different.

Technically they should be ambiguous, but realistically no one will ever do this. The analyzer will stick with the pattern of being conservative about what it reports and not handle this situation.